spec/lib/whitehall_importer/change_history_spec.rb

```diff
@@ -0,0 +1,86 @@
+RSpec.describe WhitehallImporter::ChangeHistory do
+  describe "#for" do
+    let(:first_edition) { build(:whitehall_export_edition, :superseded) }
+
+    context "when there are a list of editions" do
+      let(:second_edition) do
+        build(:whitehall_export_edition, :superseded,
+              change_note: "Some fantastic changes",
+              published_at: Time.zone.now.beginning_of_week.rfc3339)
+      end
+      let(:third_edition) do
+        build(:whitehall_export_edition, :published,
+              change_note: "Some great changes",
+              published_at: Time.zone.now.beginning_of_day.rfc3339)
+      end
+      let(:whitehall_export) do
+        build(:whitehall_export_document, editions: [first_edition, second_edition, third_edition])
+      end
+
+      it "doesn't return change history for the first edition" do
+        expect(described_class.new(whitehall_export).for(1)).to be_empty
+      end
+
+      it "doesn't return change history for the second edition" do
+        expect(described_class.new(whitehall_export).for(2)).to be_empty
+      end
+
+      it "returns change history for the third edition" do
+        expect(described_class.new(whitehall_export).for(3)).to match([
+          { "id" => anything, "note" => "Some fantastic changes", "public_timestamp" => Time.zone.now.beginning_of_week.rfc3339 },
+        ])
+      end
+
+      it "returns more change history for the fourth edition" do
+        expect(described_class.new(whitehall_export).for(4)).to match([
+          { "id" => anything, "note" => "Some great changes", "public_timestamp" => Time.zone.now.beginning_of_day.rfc3339 },
+          { "id" => anything, "note" => "Some fantastic changes", "public_timestamp" => Time.zone.now.beginning_of_week.rfc3339 },
+        ])
+      end
```

Comment:
I don't think you've really got these tests named correctly to capture the functionality of the method. They seem to be focused on very circumstantial cases.

tests I'd suggest are:

(outside of this context) it "doesn't include the first edition in change history"
it "returns the change history for the editions preceding the given edition number" 
it "returns all the change history for a number greater than the number of editions"

Reply:
You're correct, I've amended them accordingly!

Reply:
I think the thing we miss by combining those tests is any test that checks that if there are say 5 change history events and we request edition 3 we don't get back all 5 results.

Reply:
Damn I only saw this comment after I saw the approve which I merged! You make a good point which I didn't consider, I'll create another PR to add this test!